src/components/spreadsheet-view/columns/floating-treeview-list/utils/json-schema-parser.ts

```diff
@@ -35,18 +35,69 @@ function formatSpecialCases(nodeId: string): string {
     }
 
     // Handle operational limits groups with array notation
-    if (wildcardMatch('operationalLimitsGroup*', nodeId)) {
+    if (
+        wildcardMatch('operationalLimitsGroup*', nodeId) ||
+        wildcardMatch('operationalLimitsGroup*.temporaryLimitsByName.*', nodeId) ||
+        wildcardMatch('selectedOperationalLimitsGroup*.temporaryLimitsByName.*', nodeId)
+    ) {
         // Standard group1/group2 get array brackets appended
-        if (['operationalLimitsGroup1', 'operationalLimitsGroup2'].includes(nodeId)) {
+        if (
+            [
+                'operationalLimitsGroup1',
+                'operationalLimitsGroup2',
+                'operationalLimitsGroup1.temporaryLimitsByName',
+                'operationalLimitsGroup2.temporaryLimitsByName',
+                'selectedOperationalLimitsGroup1.temporaryLimitsByName',
+                'selectedOperationalLimitsGroup2.temporaryLimitsByName',
+            ].includes(nodeId)
+        ) {
             return `${nodeId}[]`;
         }
-        // Nested groups get brackets inserted before the dot
-        return nodeId.replace(/(operationalLimitsGroup\w*)(\.)/, '$1[]$2');
+
+        // Nested groups get brackets inserted before the first and the last dot
+        let replacingString = nodeId.replace(/(operationalLimitsGroup\w*)(\.)/, '$1[]$2');
+        const firstIndex = replacingString.indexOf('[].');
+        const lastIndex = replacingString.lastIndexOf('.');
+        if (lastIndex > -1 && lastIndex > firstIndex + 2) {
+            replacingString = replacingString.substring(0, lastIndex) + '[]' + replacingString.substring(lastIndex);
+        }
+        return replacingString;
```

Comment:
I'm not sure about this block. Here we want to interleave `[]` before the first dot but also after the second one in case of temporaryLimitsByName. It could be simplified by adding a second if block

```suggestion
        } else if (wildcardMatch('*temporaryLimitsByName*', nodeId)) {
            return nodeId.replace(/(operationalLimitsGroup\w*)(\.\w*)/, '$1[]$2[]');
        }
        // Nested groups get brackets inserted before the first and the last dot
        return nodeId.replace(/(operationalLimitsGroup\w*)(\.\w*)/, '$1[]$2');
```